lib/cloudkeeper/entities/convertables/convertable.rb

```diff
@@ -6,40 +6,47 @@ module Convertables
       module Convertable
         CONVERT_OUTPUT_FORMATS = %i[raw qcow2 vmdk vdi].freeze
 
-        def self.convert_output_formats
-          CONVERT_OUTPUT_FORMATS
-        end
-
-        FORMAT_REGEX = /^to_(?<format>#{convert_output_formats.join('|')})$/
-
         def self.included(base)
           raise Cloudkeeper::Errors::Convertables::ConvertabilityError, "#{base.inspect} cannot become a convertable" \
             unless base.method_defined?(:file) && base.method_defined?(:format)
 
           super
         end
 
+        def convert_output_formats
+          CONVERT_OUTPUT_FORMATS
+        end
+
+        def format_regex
+          /^to_(?<format>#{convert_output_formats.join('|')})$/
+        end
+
         def method_missing(method, *arguments, &block)
-          result = method.to_s.match(FORMAT_REGEX)
-          return convert(result[:format]) if result && result[:format]
+          result = method.to_s.match(format_regex)
+          if result && result[:format]
+            return self if format.to_sym == result[:format].to_sym
+            return convert result[:format]
+          end
 
           super
         end
 
         def respond_to_missing?(method, *)
-          method =~ FORMAT_REGEX || super
+          method =~ format_regex || super
         end
 
         private
 
         def convert(output_format)
           logger.debug "Converting file #{file.inspect} from #{format.inspect} to #{output_format.inspect}"
-          return self if output_format.to_sym == format.to_sym
```

Comment:
It was just moved here https://github.com/the-cloudkeeper-project/cloudkeeper/pull/28/files#diff-fd45d5eb6f238bc701e14fbfb268c4beR27
I can see your reviews are thorough :smirk: